modules/database/src/ioc/db/dbAccess.c

```diff
@@ -1332,25 +1345,33 @@ long dbPut(DBADDR *paddr, short dbrType,
 
     if (paddr->pfldDes->special == SPC_DBADDR &&
         prset && prset->get_array_info) {
-        long dummy;
 
-        status = prset->get_array_info(paddr, &dummy, &offset);
+        status = prset->get_array_info(paddr, &available_no_elements, &offset);
         /* paddr->pfield may be modified */
         if (status) goto done;
-        if (no_elements < nRequest)
-            nRequest = no_elements;
```

Comment:
I guess this bounds limit is now expected to be enforced elsewhere?

Reply:
Yes, it is done now immediately before the call to `dbPutConvertRoutine` (for the other cases it is not needed).